crypto/asn1/i2d_evp.c

```diff
@@ -119,6 +124,16 @@ int i2d_PrivateKey(const EVP_PKEY *a, unsigned char **pp)
     return -1;
 }
 
+int i2d_PrivateKey(const EVP_PKEY *a, unsigned char **pp)
+{
+    return i2d_PrivateKey_impl(a, pp, 1);
+}
+
+int i2d_PKCS8PrivateKey(const EVP_PKEY *a, unsigned char **pp)
```

Comment:
I might have expected this to be implemented using `do_pk8pkey` in crypto/pem/pem_pk8.c where all the rest of the `i2d_PKCS8PrivateKey` functions are implemented.

Reply:
But, as you see from the implementation, this is the existing `i2d_PrivateKey` with the logic for traditional format skipped over.

Reply:
In particular, no need for an intermediate BIO, `isder` boolean, cipher, key or passphrase arguments...